central_v2/src/components/button/favouritebutton/FavouriteButton.tsx

```diff
@@ -19,14 +18,17 @@ export default function FavouriteButton({
 }: FavouriteButtonProps) {
   const [isLoading, setIsLoading] = useState<boolean>(false);
   const apiClients = useTSAPIClientsContext(APIClientsContext);
-  const userProfile = useUserProfileContext(UserProfileContext);
-  const userProfileDispatch = useUserProfileDispatchContext(UserProfileDispatchContext);
-  const isFavorite = userProfile?.favoriteGameTemplateIds?.includes(id) ?? false;
+  const centralData = useCentralDataState();
```

Comment:
this is what it looks like to use the context down in the component tree. You can think of this like 

const `[state, setState] = useState();`

where `centralData` is the equivalent of `state` and `centralDataDispatch` is the equivalent of `setState`